### PR TITLE
[1.12] Fix validation of overlay backends

### PR DESCRIPTION
**Pull request by urbanserj**
## High-level description

test_endpoints.test_if_overlay_master_agent_is_up fails for 1.11 upgraded clusters.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS-46506](https://jira.mesosphere.com/browse/DCOS-46506) test_endpoints.test_if_overlay_master_agent_is_up fails for 1.11 upgraded clusters

## Related tickets (optional)

Other tickets related to this change:

## Checklist for all PRs

  - [x] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change: it's a fix for integration tests.
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: none
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by urbanserj** [comment]
@mesosphere-mergebot label [Ready For Review]

File: packages/dcos-integration-test/extra/test_endpoints.py [hunk 1]
```diff
@@ -315,26 +315,25 @@ def _validate_overlay_subnet(agent_subnet, overlay_subnet, prefixlen):
 def _validate_overlay_backend(overlay_name, backend):
     try:
         # Make sure the backend has the right VNI.
-        vxlan = backend.pop('vxlan')
+        vxlan = backend['vxlan']
```
Comment (orlandohohmeier) [inline packages/dcos-integration-test/extra/test_endpoints.py]: 🤔 Wonder why used `backend.pop('vxlan')` ... I can't find any place where we seem to rely on this test mutating the dictionary but feel uncomfortable green-lighting the PR as is without further investigation esp. as we seem to access quite some elements using `pop` in this file.

Comment (GoelDeepak) [reply packages/dcos-integration-test/extra/test_endpoints.py]: `pop` was validating two things:
1. The object that we want to validate
2. The overall structure of the backend

The change has taken out the second test. And, so if there are some extra key-value pairs in the backend json, it will go unvalidated. Maybe, it was a good thing to validate the backend structure as well.

Comment (Fabs) [reply packages/dcos-integration-test/extra/test_endpoints.py]: I wonder if it would be possible to validate it without mutating the `backend`? Asserting something on the array of keys maybe?

Comment (orlandohohmeier) [reply packages/dcos-integration-test/extra/test_endpoints.py]: @GoelDeepak I wasn't referring to the `vxlan.pop('vtep_*')` but `backend.pop('vxlan')` call. I understand that we were removing the `vtep_*`  elements and then comparing the rest to an `expected_vxlan` dictionary verifying that only the specified elements were present and that we're now no longer failing the validation in case there are _unknown_ elements which I would argue is ok. I'm also very much in support of dropping the `pop` calls as I'd argue that they make it uncessary hard to follow the code.

I've further reviewed the flow and couldn't spot any place where we're relying on backend being mutated.
